client/components/sub-masterbar-nav/dropdown.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'gridicons';
+import Item from './item';
+
+export default class Dropdown extends Component {
+
+	static propTypes = {
+		children: PropTypes.func.isRequired,
```

Comment:
I have hard time following this. I will explain my line of thoughts on how I would see it so you can comment.
I would rather not pass callback that generates children but the actual data that are used to generate children. Then I would implement function that generates children ( Item ) in both components even if it would mean duplication - but in that case not that big. Why? 

* It is hard to figure out ( not that bad here but it is a general remark ) how the passed props are used.
You pass children callback to this component and it is used here. At the same time you pass `onSelect` into the callback, and to understand how it will be used you need to back to the calling component that is the origin of `children` callback.

* This is I think more important: we loose all benefits of smart optimizations of React lifecycle methods. If in calling component set of items will change then `shouldComponentUpdate` here will only see the method in props that will not change. Right now it is not the case here but we don't know how this component will end up.

* Last thing I would not use children here. I would rather think in terms of composition.